decidim-assemblies/spec/system/assemblies_spec.rb

```diff
@@ -36,6 +37,53 @@
     switch_to_host(organization.host)
   end
 
+  describe "testing image hierarchy for meta tags" do
+    let!(:assembly) { create(:assembly, organization:, description:, hero_image:, banner_image:) }
+    let(:banner_image) { nil }
+    let(:hero_image) { nil }
+    let(:description) { { en: "Description <p><img src=\"#{description_image_path}\"></p>" } }
+
+    context "when the assembly has an attachment" do
+      let!(:assembly_attachment) { create(:attachment, :with_image, attached_to: assembly, file: assembly_attachment_file) }
+      let!(:assembly_attachment_file) { Decidim::Dev.test_file("city2.jpeg", "image/jpeg") }
+
+      it_behaves_like "meta image url examples", "city2.jpeg" do
+        let(:resource) { assembly }
+      end
+    end
+
+    context "when assembly attachment is not present and the description image is there" do
+      it_behaves_like "meta image url examples", "description_image.jpg" do
+        let(:resource) { assembly }
+      end
+    end
+
+    context "when neither assembly attachment nor description image is present" do
+      let(:file) { Decidim::Dev.test_file("city3.jpeg", "image/jpeg") }
+      let!(:assembly_hero_attachment) { create(:attachment, :with_image, attached_to: assembly, file:) }
+      let(:hero_image) { Decidim::Dev.test_file("city3.jpeg", "image/jpeg") }
```

Comment:
```suggestion
      let(:file) { Decidim::Dev.test_file("city2.jpeg", "image/jpeg") }
      let!(:assembly_hero_attachment) { create(:attachment, :with_image, attached_to: assembly, file:) }
      let(:hero_image) { Decidim::Dev.test_file("city3.jpeg", "image/jpeg") }
```

You are using the same image here, makes it impossible to know which one is being used!